src/snowflake/snowpark/mock/_functions.py

```diff
@@ -884,20 +884,20 @@ def mock_to_binary(
 ) -> ColumnEmulator:
     """
     [x] TO_BINARY( <string_expr> [, '<format>'] )
-    [ ] TO_BINARY( <variant_expr> )
+    [x] TO_BINARY( <variant_expr> )
     """
-    if isinstance(column.sf_type.datatype, (StringType, NullType)):
-        fmt = fmt.upper() if fmt else "HEX"
-        if fmt == "HEX":
-            res = column.apply(lambda x: try_convert(binascii.unhexlify, try_cast, x))
-        elif fmt == "BASE64":
-            res = column.apply(lambda x: try_convert(base64.b64decode, try_cast, x))
-        elif fmt == "UTF-8":
-            res = column.apply(
-                lambda x: try_convert(lambda y: y.encode("utf-8"), try_cast, x)
-            )
-        else:
-            raise SnowparkSQLException(f"Invalid binary format {fmt}")
+    fmt = fmt.upper() if fmt else "HEX"
+    fmt_decoder = {
+        "HEX": binascii.unhexlify,
+        "BASE64": base64.b64decode,
+        "UTF-8": lambda x: x.encode("utf-8"),
```

Comment:
nit: do you think it will help other function implementation by turning this into a module variable so that it can be shared?

Reply:
Let's keep it here for now. We can move it to utils as a module level constant if this is ever reused.